common/wonder_upgrades/wc_wonders_historical_upgrades.txt

```diff
@@ -1 +1,69 @@
 #Last Modified: 7/11/2019
+
+upgrade_forge_of_wills = {
+	picture = GFX_upgrade_icon_forge_of_wills
+
+	potential = { }
+	allow = { }
+	active = { }
+
+	flags = {
+		is_singular
+	}
+	build_time = 24
+	gold_cost_flat = 400
+	gold_cost_ticking = 1
+	local_modifier = {
+		garrison_size = 1
+	}
+	owner_modifier = {
+		retinuesize = 250
+		retinuesize_perc = 0.2
+		text_effect_titanforged_retinue = 1
+	}
+}
+upgrade_silithid_greathive = {
+	picture = GFX_upgrade_icon_silithid_greathive
+
+	potential = { }
+	allow = { }
+	active = { }
+
+	flags = {
+		is_singular
+	}
+	build_time = 24
+	gold_cost_flat = 400
+	gold_cost_ticking = 1
+	local_modifier = {
+		levy_reinforce_rate = 0.25
+	}
+	owner_modifier = {
+		retinuesize = 250
+		retinuesize_perc = 0.2
+		text_effect_qiraji_retinue = 1
+	}
+}
+upgrade_crucible_of_storms = {
+	picture = GFX_upgrade_icon_crucible_of_storms
+
+	potential = { }
+	allow = { religion = old_gods_worship }
+	active = { religion = old_gods_worship }
+
+	flags = {
+		is_singular
+	}
+	build_time = 24
+	gold_cost_flat = 400
+	gold_cost_ticking = 1
+	local_modifier = {
+	}
+	owner_modifier = {
+		land_organisation = 0.05
+		retinuesize = 250
+		retinuesize_perc = 0.2
+		text_effect_void_inflow = 1
+		text_effect_naga_retinue = 1
+	}
```

Comment:
The best upgrade. Gives everything that other upgrades give + Void Inflow + `land_organisation`.

@DukeJeans had [Documentation](https://github.com/Warcraft-GoA-Development-Team/Warcraft-Guardians-of-Azeroth/wiki/Links-to-External-Documentation) to balance wonders, check it out. Not sure how it works though. Maybe it's better to use the balancing codex at this point.

Reply:
Check the other wonder upgrades again. They have local modifier buffs

Reply:
The problem that they're local while `land_organisation` affects more than one province/army/character, etc. So, we need to come up with some balance.

Reply:
I feel it like `land_organisation` is too strong and `text_effect_void_inflow` more or less (most likely more) costs like local modifiers of other wonders.

Reply:
Getting +100% garrison on your old god province is much better than void inflow imo

Reply:
And land organization and void inflow are from one of wonder_upgrades

Reply:
> Getting +100% garrison on your old god province is much better than void inflow imo

Maybe you're right, void inflow gives a lot, but it's random. But `garrison_size` is on the same level, to be honest. It just makes you sure the enemy needs more troops to assault your capital.

Reply:
well the doc by @DukeJeans shows that my modifiers are pretty balanced.

Reply:
Hmm? I don't see them in the doc.

Reply:
I used the doc to look at what modifiers are used there